### PR TITLE
ADBDEV-6699: Fix tests for GPDB 7

### PR DESCRIPTION
**Pull request by silent-observer**
Fix tests for GPDB 7

The tests failed if it ran with GPDB 7. This patch makes the following changes:

1. In `test_create_extension` (both in regress and isolation2) use
`default_table_access_method` GUC instead of `gp_default_storage_options`
since `gp_default_storage_options` can't be used for appendonly in GPDB 7.
Since we can't choose which one to use, we run both, expecting one of them to
fail.
2. In `test_activetable_limit` and `test_rejectmap_limit` (regress) change log
reading function to read from either `pg_logs` or `logs`, depending on which
one is available.
3. In `test_dropped_table` and `test_temporary_table` (isolation2) add
matchsubs to account for different command output on GPDB 7.
4. Update README to indicate support for GPDB 7 on Ubuntu 22.

### CODE REVIEW

File: tests/regress/sql/test_activetable_limit.sql [hunk 1]
```diff
@@ -22,7 +22,11 @@ INSERT INTO a03 values(generate_series(0, 500));
 CREATE EXTENSION diskquota;
 -- we only read the current log file
 CREATE EXTERNAL WEB TABLE segment_logs(line text)
-    EXECUTE 'cat $GP_SEG_DATADIR/pg_log/$(ls -Art $GP_SEG_DATADIR/pg_log | tail -n 1)'
+    EXECUTE 'if [ -d "$GP_SEG_DATADIR/pg_log" ]; then 
+            cat $GP_SEG_DATADIR/pg_log/$(ls -Art $GP_SEG_DATADIR/pg_log | tail -n 1)
+        else
+            cat $GP_SEG_DATADIR/log/$(ls -Art $GP_SEG_DATADIR/log | tail -n 1)
+        fi'
```
Comment (RekGRpth) [inline tests/regress/sql/test_activetable_limit.sql:29]: why not using here `gp_toolkit.__gp_log_segment_ext` and below `gp_toolkit.__gp_log_master_ext` and filter them like
```sql
join pg_stat_activity a on logsession = 'con'||sess_id where pid = pg_backend_pid() and logmessage like '%the number of active tables reached the limit%'
```

File: tests/regress/expected/test_create_extension.out [hunk 2]
```diff
@@ -1,8 +1,16 @@
 -- check that diskquota ignores global flag appendonly=true
+-- for GPDB 6
 SET gp_default_storage_options='appendonly=true';
+-- for GPDB 7
+SET default_table_access_method='ao_row';
+ERROR:  unrecognized configuration parameter "default_table_access_method"
```
Comment (RekGRpth) [inline tests/regress/expected/test_create_extension.out]: why not
```sql
select case
    when current_setting('server_version_num')::int > 120000
    then set_config('default_table_access_method', 'ao_row', false)
    else set_config('gp_default_storage_options', 'appendonly=true', false)
end;
```

File: arenadata/README.md [hunk 3]
```diff
@@ -6,23 +6,23 @@ How to build Docker image: (["readme.md"](https://github.com/arenadata/gpdb/blob
 1. Download the cmake-3.20 install script from ([source](https://github.com/Kitware/CMake/releases/download/v3.20.0/cmake-3.20.0-linux-x86_64.sh)).
 
 2. Build diskquota in the Docker container.
-Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine.
+Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to GPDB regression testing image (for example `hub.adsw.io/library/gpdb6_regress:latest` or `hub.adsw.io/library/gpdb7_u22:latest`)
```
Comment (RekGRpth) [inline arenadata/README.md]:
```suggestion
Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to GPDB regression testing image (for example `hub.adsw.io/library/gpdb6_u22:latest` or `hub.adsw.io/library/gpdb7_u22:latest`)
```

File: arenadata/README.md [hunk 4]
```diff
@@ -6,23 +6,23 @@ How to build Docker image: (["readme.md"](https://github.com/arenadata/gpdb/blob
...
        -v /tmp/diskquota_artifacts:/home/gpadmin/diskquota_artifacts \
        -v <PATH_TO_DISKQUOTA_SRC>:/home/gpadmin/diskquota_src \
        -v <PATH_TO_CMAKE_INSTALL_SCRIPT>:/home/gpadmin/bin_cmake/cmake-3.20.0-linux-x86_64.sh \
-       hub.adsw.io/library/gpdb6_regress:latest diskquota_src/concourse/scripts/entry.sh build
+       <IMAGE> diskquota_src/concourse/scripts/entry.sh build
 ```
 
 3. Run tests.
-Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine.
+Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to GPDB regression testing image (for example `hub.adsw.io/library/gpdb6_regress:latest` or `hub.adsw.io/library/gpdb7_u22:latest`)
```
Comment (RekGRpth) [inline arenadata/README.md]:
```suggestion
Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to GPDB regression testing image (for example `hub.adsw.io/library/gpdb6_u22:latest` or `hub.adsw.io/library/gpdb7_u22:latest`)
```

File: tests/regress/sql/test_activetable_limit.sql [hunk 5]
```diff
@@ -34,7 +30,7 @@ SELECT diskquota.wait_for_worker_new_epoch();
 CREATE TABLE s.t1 (a int, b int) DISTRIBUTED BY (a)
     PARTITION BY RANGE (b) ( START (0) END (10) EVERY (1) );
 
-SELECT count(*) FROM segment_logs WHERE line LIKE '%the number of active tables reached the limit%';
+SELECT count(*) FROM gp_toolkit.__gp_log_segment_ext JOIN pg_stat_activity a ON logsession = 'con'||sess_id WHERE pid = pg_backend_pid() AND logmessage LIKE '%the number of active tables reached the limit%';
```
Comment (RekGRpth) [inline tests/regress/sql/test_activetable_limit.sql]:
```suggestion
SELECT count(*) FROM gp_toolkit.__gp_log_segment_ext JOIN pg_stat_activity ON logsession = 'con'||sess_id WHERE pid = pg_backend_pid() AND logmessage LIKE '%the number of active tables reached the limit%';
```

File: tests/regress/sql/test_activetable_limit.sql [hunk 6]
```diff
@@ -47,7 +43,7 @@ SELECT count(*) FROM s.t1;
 -- altered reloid cache overflow check. expected warning.
 VACUUM FULL;
 
-SELECT count(*) FROM segment_logs WHERE line LIKE '%the number of altered reloid cache entries reached the limit%';
+SELECT count(*) FROM gp_toolkit.__gp_log_segment_ext JOIN pg_stat_activity a ON logsession = 'con'||sess_id WHERE pid = pg_backend_pid() AND logmessage LIKE '%the number of altered reloid cache entries reached the limit%';
```
Comment (RekGRpth) [inline tests/regress/sql/test_activetable_limit.sql]:
```suggestion
SELECT count(*) FROM gp_toolkit.__gp_log_segment_ext JOIN pg_stat_activity ON logsession = 'con'||sess_id WHERE pid = pg_backend_pid() AND logmessage LIKE '%the number of altered reloid cache entries reached the limit%';
```

File: tests/regress/sql/test_rejectmap_limit.sql [hunk 7]
```diff
@@ -44,7 +40,7 @@ INSERT INTO s5.a SELECT generate_series(1,100000);
 
 SELECT diskquota.wait_for_worker_new_epoch();
 
-SELECT count(*) FROM master_log WHERE line LIKE '%the number of local quota reject map entries reached the limit%' AND line NOT LIKE '%LOG%';
+SELECT count(*) FROM gp_toolkit.__gp_log_master_ext WHERE logmessage LIKE '%the number of local quota reject map entries reached the limit%' AND logmessage NOT LIKE '%LOG%';
```
Comment (RekGRpth) [inline tests/regress/sql/test_rejectmap_limit.sql]:
```suggestion
SELECT count(*) FROM gp_toolkit.__gp_log_master_ext JOIN pg_stat_activity ON logsession = 'con'||sess_id WHERE pid = pg_backend_pid() AND logmessage LIKE '%the number of local quota reject map entries reached the limit%' AND logmessage NOT LIKE '%LOG%';
```

Comment (silent-observer) [reply tests/regress/sql/test_rejectmap_limit.sql]: Doesn't work, the session printing the warning here is somehow different from the current session. Still, the behavior should be identical to existing one (which simply reads the last line in the log file)

File: tests/isolation2/expected/test_dropped_table.out [hunk 8]
```diff
@@ -1,5 +1,18 @@
...
+-- s/CREATE SCHEMA/CREATE/
+-- m/CREATE TABLE/
+-- s/CREATE TABLE/CREATE/
+-- m/INSERT 0 \d+/
+-- s/INSERT 0 (\d+)/INSERT $1/
+-- m/DROP SCHEMA/
+-- s/DROP SCHEMA/DROP/
+-- m/DROP TABLE/
+-- s/DROP TABLE/DROP/
+-- end_matchsubs
```
Comment (red1452) [inline tests/isolation2/expected/test_dropped_table.out]: I think, it should be in the `init_file`.

File: arenadata/README.md [hunk 9]
```diff
@@ -6,23 +6,23 @@ How to build Docker image: (["readme.md"](https://github.com/arenadata/gpdb/blob
...
        -v /tmp/diskquota_artifacts:/home/gpadmin/diskquota_artifacts \
        -v <PATH_TO_DISKQUOTA_SRC>:/home/gpadmin/diskquota_src \
        -v <PATH_TO_CMAKE_INSTALL_SCRIPT>:/home/gpadmin/bin_cmake/cmake-3.20.0-linux-x86_64.sh \
-       hub.adsw.io/library/gpdb6_regress:latest diskquota_src/concourse/scripts/entry.sh build
+       <IMAGE> diskquota_src/concourse/scripts/entry.sh build
 ```
 
 3. Run tests.
-Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine.
+Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to GPDB regression testing image (for example `hub.adsw.io/library/gpdb6_u22:latest` or `hub.adsw.io/library/gpdb7_u22:latest`)
```
Comment (bandetto) [inline arenadata/README.md]: Maybe avoid duplication of this statement (and this statement at all), add a list of currently supported images for disquota somewhere above, and just refer to it?
```suggestion
Change <PATH_TO_DISKQUOTA_SRC> and <PATH_TO_CMAKE_INSTALL_SCRIPT> to the appropriate paths on your local machine, and <IMAGE> to a supported GPDB docker image.
```

File: tests/regress/sql/test_activetable_limit.sql (hunk 1 above)
Comment (bandetto) [reply tests/regress/sql/test_activetable_limit.sql:29]: Can you please make a Github suggestion out of this change?

File: arenadata/README.md [hunk 10]
```diff
@@ -3,26 +3,33 @@
 You can build your Docker image from GPDB source or use prebuilt images from hub.adsw.io.
 How to build Docker image: (["readme.md"](https://github.com/arenadata/gpdb/blob/f7ff7c8ecae4ce7ab3b73fd46171cdaa457b3591/arenadata/readme.md)).
 
+## Supported GPDB images
+
+- `hub.adsw.io/library/gpdb6_u22:latest`
```
Comment (RekGRpth) [inline arenadata/README.md:9]:
```suggestion
- `hub.adsw.io/library/gpdb6_regress:latest`
- `hub.adsw.io/library/gpdb6_u22:latest`
```

File: tests/regress/sql/test_activetable_limit.sql (hunk 1 above)
Comment (bandetto) [reply tests/regress/sql/test_activetable_limit.sql:29]: I tested this and came to a conclusion that receiving the last log message from each segment using SQL will require a more clunkier code than this way.

**Review by bandetto — APPROVED** [review]

**Review by red1452 — COMMENTED** [review]
The tests were failing if run with GPDB 7 -> The tests failed if it ran with GPDB 7

**Review by red1452 — APPROVED** [review]

**Pull request merged**